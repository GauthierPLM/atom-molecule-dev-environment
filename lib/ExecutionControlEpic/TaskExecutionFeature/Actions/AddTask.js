'use babel'
// @flow
import type {Task, DevToolInfos} from "../Types/types.js.flow";

export type AddTaskAction = {
  type: 'ADD_TASK',
  payload: {
    task: Task
  };
};

export function addTask(id: string, target: {config: mixed, tool: DevToolInfos}): AddTaskAction {
  return {
    type: 'ADD_TASK',
    payload: {
      task: {
        id,
        target,
        state: "created",
      }
    }
  };
};
