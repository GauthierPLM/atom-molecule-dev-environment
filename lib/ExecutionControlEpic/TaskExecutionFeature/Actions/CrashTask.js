'use babel'
// @flow

export type CrashTaskAction = {
  type: 'CRASH_TASK',
  payload: {
    id: string,
    date: number
  };
};

export function crashTask(id: string, date: number): CrashTaskAction {
  return {
    type: 'CRASH_TASK',
    payload: {
      id,
      date
    }
  };
};
