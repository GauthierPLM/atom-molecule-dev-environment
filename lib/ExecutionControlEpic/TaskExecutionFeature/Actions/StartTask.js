"use babel";
// @flow

export type StartTaskAction = {
  type: "START_TASK",
  payload: {
    id: string,
  },
};

export function startTask(id: string): StartTaskAction {
  return {
    type: "START_TASK",
    payload: {
      id,
    },
  };
}
