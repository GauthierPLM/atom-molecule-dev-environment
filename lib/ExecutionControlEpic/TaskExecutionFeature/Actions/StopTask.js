"use babel";
// @flow

export type StopTaskAction = {
  type: "STOP_TASK",
  payload: {
    id: string,
    date: number,
  },
};

export function stopTask(id: string, date: number): StopTaskAction {
  return {
    type: "STOP_TASK",
    payload: {
      id,
      date,
    },
  };
}
