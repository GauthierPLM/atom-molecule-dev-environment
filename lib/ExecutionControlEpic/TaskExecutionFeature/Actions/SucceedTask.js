"use babel";
// @flow

export type SucceedTaskAction = {
  type: "SUCCEED_TASK",
  payload: {
    id: string,
    date: number
  }
};

export function succeedTask(id: string, date: number): SucceedTaskAction {
  return {
    type: "SUCCEED_TASK",
    payload: {
      id,
      date
    }
  };
}
