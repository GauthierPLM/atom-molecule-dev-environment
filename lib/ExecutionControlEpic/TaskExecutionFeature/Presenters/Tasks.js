"use babel";
// @flow

import React from "react";
import Task from "./Task";

export default class Tasks extends React.Component<DefaultProps, Props, State> {
  state: State;
  props: Props;
  static defaultProps: DefaultProps;

  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <div style={{ overflow: "auto", flex: "1", display: "inline-flex" }}>
        <ul style={{ padding: "0px" }}>
          {this.props.tasks
            .slice(0, this.props.limited ? 5 : undefined)
            .map(task => (
              <li
                style={{
                  listStyle: "none",
                  opacity: task.selected ? "1" : "0.5",
                }}
                key={task.id}
              >
                <Task {...task} onClick={() => this.props.onTaskClick(task)} />
              </li>
            ))}
        </ul>
      </div>
    );
  }
}

Tasks.defaultProps = {
  limited: false,
};

type DefaultProps = {
  limited: boolean,
};

type Props = {
  limited: boolean,
  onTaskClick(task: Task): void,
  tasks: Array<Task & { selected: boolean }>,
};

type State = {};
