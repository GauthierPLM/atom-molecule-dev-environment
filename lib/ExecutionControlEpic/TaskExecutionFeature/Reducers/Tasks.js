'use babel'
// @flow
import type {Task} from "../Types/types.js.flow";
import { fromJS } from "immutable";

export default function(state: TasksReducer = [], action: any): TasksReducer {
  switch(action.type) {
    case "ADD_TASK":
      return fromJS(state)
              .push(action.payload.task)
              .toJS();
    case "START_TASK":
      return fromJS(state)
              .map(task => task.get('id') == action.payload.id ?
                    task.update(t => t.get('state') == "created" ?
                      t.set('state', "running")
                       .set('end', action.payload.date)
                     : t) : task
              )
              .toJS();
    case "STOP_TASK":
      return fromJS(state)
              .map(task => task.get('id') == action.payload.id ?
                    task.update(t => t.get('state') == "running" ?
                      t.set('state', "stopped")
                       .set('end', action.payload.date)
                     : t) : task
              )
              .toJS();
    case "CRASH_TASK":
      return fromJS(state)
              .map(task => task.get('id') == action.payload.id ?
                    task.update(t => t.get('state') == "running" ?
                      t.set('state', "crashed")
                       .set('end', action.payload.date)
                     : t) : task
              )
              .toJS();
    case "FAIL_TASK":
      return fromJS(state)
              .map(task => task.get('id') == action.payload.id ?
                    task.update(t => t.get('state') == "running" ?
                      t.set('state', "failed")
                       .set('end', action.payload.date)
                     : t) : task
              )
              .toJS();
    case "SUCCEED_TASK":
      return fromJS(state)
              .map(task => task.get('id') == action.payload.id ?
                    task.update(t => t.get('state') == "running" ?
                      t.set('state', "succeed")
                       .set('end', action.payload.date)
                     : t) : task
              )
              .toJS();
    default:
      return state;
  }
}

export type TasksReducer = Array<Task>;
