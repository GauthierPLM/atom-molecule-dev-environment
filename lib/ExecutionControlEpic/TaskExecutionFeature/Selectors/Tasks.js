'use babel'
// @flow

import type {TaskState} from "../Types/types.js";
import { fromJS, is } from "immutable";
import type {TasksReducer} from "../Reducers/Tasks";

function selectRepresentativeState(tasks: any) {
  if (tasks.find(task => task.get('state') == "running"))
    return "running";
  else {
    return tasks.reduce((red, task) => {
      if (red.get('state') == null)
        return task;
      else if (red.get('end') && task.get('end'))
        return red.get('end') < task.get('end') ? task : red;
      else if (task.get('end'))
        return task;
      else
        return red;
    }, fromJS({state: null})).get('state');
  };
}

export function selectStateOfTarget(state: TasksReducer, target: mixed): TaskState {
  return fromJS(state)
          .filter(task => is(task.get('target'), fromJS(target)))
          .update(selectRepresentativeState);
};

export function selectStateOfTool(state: TasksReducer, toolId: string): TaskState {
  return fromJS(state)
          .filter(task => task.getIn(['target', 'tool', 'id']) == toolId)
          .update(selectRepresentativeState);
};
