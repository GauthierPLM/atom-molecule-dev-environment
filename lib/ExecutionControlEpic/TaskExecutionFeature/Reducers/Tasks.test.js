'use babel'
// @flow

import Tasks from "./Tasks";
import type {TasksReducer} from "./Tasks";
import type {AddTaskAction} from "../Actions/AddTask";
import {addTask} from "../Actions/AddTask";
import type {CrashTaskAction} from "../Actions/CrashTask";
import {crashTask} from "../Actions/CrashTask";
import type {StopTaskAction} from "../Actions/StopTask";
import {stopTask} from "../Actions/StopTask";
import type {StartTaskAction} from "../Actions/StartTask";
import {startTask} from "../Actions/StartTask";
import type {SucceedTaskAction} from "../Actions/SucceedTask";
import {succeedTask} from "../Actions/SucceedTask";
import type {FailTaskAction} from "../Actions/FailTask";
import {failTask} from "../Actions/FailTask";

describe('Tasks Reducer', () => {
  describe('ADD_TASK', () => {
    it('should add a task', () => {
      let action: AddTaskAction = addTask("1", {config: 2, tool: {id: "3", name: "tool", iconUri: ""}}, 1475954335);
      let state: TasksReducer = [];
      let subject = Tasks(state, action);

      expect(subject).toEqual([action.payload.task]);
    });

  });

  describe('START_TASK', () => {
    it('should mark the task as started', () => {
      let action: StartTaskAction = startTask("2");
      let state: TasksReducer = [{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }];
      let subject = Tasks(state, action);

      expect(subject).toEqual([{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "running",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }]);
    });

  });

  describe('STOP_TASK', () => {
    it('should mark the task as stopped', () => {
      let action: StopTaskAction = stopTask("2", 1475954336);
      let state: TasksReducer = [{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "running",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }];
      let subject = Tasks(state, action);

      expect(subject).toEqual([{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "stopped",
        debut: 1475954335,
        end: 1475954336,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }]);
    });
  });

  describe('CRASH_TASK', () => {
    it('should mark the task as crashed', () => {
      let action: CrashTaskAction = crashTask("2", 1475954336);
      let state: TasksReducer = [{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "running",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }];
      let subject = Tasks(state, action);

      expect(subject).toEqual([{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "crashed",
        debut: 1475954335,
        end: 1475954336,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }]);
    });
  });

  describe('SUCCEED_TASK', () => {
    it('should mark the task as succeed', () => {
      let action: SucceedTaskAction = succeedTask("2", 1475954336);
      let state: TasksReducer = [{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "running",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }];
      let subject = Tasks(state, action);

      expect(subject).toEqual([{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "succeed",
        debut: 1475954335,
        end: 1475954336,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }]);
    });
  });

  describe('FAIL_TASK', () => {
    it('should mark the task as failed', () => {
      let action: FailTaskAction = failTask("2", 1475954336);
      let state: TasksReducer = [{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "running",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }];
      let subject = Tasks(state, action);

      expect(subject).toEqual([{
        id: "1",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "2",
        state: "failed",
        debut: 1475954335,
        end: 1475954336,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }, {
        id: "3",
        state: "created",
        debut: 1475954335,
        target: {
          config: 2,
          tool: {
            id: "1",
            name: "tool",
            iconUri: ""
          }
        }
      }]);
    });
  });
});
