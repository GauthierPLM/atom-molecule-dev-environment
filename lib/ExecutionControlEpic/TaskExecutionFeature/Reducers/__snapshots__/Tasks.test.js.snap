// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks Reducer CRASH_TASK should mark the task as crashed 1`] = `
Array [
  Object {
    "debut": 1475954335,
    "id": "1",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "end": 1475954336,
    "id": "2",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "crashed",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "id": "3",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
]
`;

exports[`Tasks Reducer FAIL_TASK should mark the task as failed 1`] = `
Array [
  Object {
    "debut": 1475954335,
    "id": "1",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "end": 1475954336,
    "id": "2",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "failed",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "id": "3",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
]
`;

exports[`Tasks Reducer START_TASK should mark the task as started 1`] = `
Array [
  Object {
    "debut": 1475954335,
    "id": "1",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "end": undefined,
    "id": "2",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "running",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "id": "3",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
]
`;

exports[`Tasks Reducer STOP_TASK should mark the task as stopped 1`] = `
Array [
  Object {
    "debut": 1475954335,
    "id": "1",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "/ok",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "end": 1475954336,
    "id": "2",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "stopped",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "/ok",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "id": "3",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "/ok",
      "type": "shell",
    },
  },
]
`;

exports[`Tasks Reducer STOP_TASK_OF_TARGET should mark the task as stopped 1`] = `
Array [
  Object {
    "debut": 1475954335,
    "id": "1",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "end": 1475954336,
    "id": "2",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "2",
        "name": "tool",
      },
    },
    "state": "stopped",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "id": "3",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
]
`;

exports[`Tasks Reducer SUCCEED_TASK should mark the task as succeed 1`] = `
Array [
  Object {
    "debut": 1475954335,
    "id": "1",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "end": 1475954336,
    "id": "2",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "succeed",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
  Object {
    "debut": 1475954335,
    "id": "3",
    "plan": Object {
      "config": 2,
      "packageInfos": Object {
        "name": "root",
        "path": "/",
        "type": "file",
      },
      "stager": Object {
        "host": null,
        "type": "integrated",
      },
      "tool": Object {
        "iconUri": "",
        "id": "1",
        "name": "tool",
      },
    },
    "state": "created",
    "step": 1,
    "strategy": Object {
      "command": "",
      "cwd": "",
      "type": "shell",
    },
  },
]
`;
